src/gnome/dialog-imap-editor.c

```diff
@@ -0,0 +1,759 @@
+    GtkWidget    *filter_button;
+    GtkWidget    *filter_text_entry;
+    GtkWidget    *filter_label;
+
+    GtkWidget    *expand_button;
+    GtkWidget    *collapse_button;
+}ImapDialog;
+
+
+/* This static indicates the debugging module that this .o belongs to.  */
+static QofLogModule log_module = GNC_MOD_GUI;
+
+void gnc_imap_dialog_window_destroy_cb (GtkWidget *object, gpointer user_data);
+void gnc_imap_dialog_close_cb (GtkDialog *dialog, gpointer user_data);
+void gnc_imap_dialog_response_cb (GtkDialog *dialog, gint response_id, gpointer user_data);
+
+static void get_account_info (ImapDialog *imap_dialog);
+
+void
+gnc_imap_dialog_window_destroy_cb (GtkWidget *object, gpointer user_data)
+{
+    ImapDialog *imap_dialog = user_data;
+
+    ENTER(" ");
+    gnc_unregister_gui_component_by_data (DIALOG_IMAP_CM_CLASS, imap_dialog);
+
+    if (imap_dialog->dialog)
+    {
+        gtk_widget_destroy (imap_dialog->dialog);
+        imap_dialog->dialog = NULL;
+    }
+    g_free (imap_dialog);
+    LEAVE(" ");
+}
+
+void
+gnc_imap_dialog_close_cb (GtkDialog *dialog, gpointer user_data)
+{
+    ImapDialog *imap_dialog = user_data;
+
+    ENTER(" ");
+    gnc_close_gui_component_by_data (DIALOG_IMAP_CM_CLASS, imap_dialog);
+    LEAVE(" ");
+}
+
+static gboolean
+are_you_sure (ImapDialog *imap_dialog)
+{
+    GtkWidget   *dialog;
+    gint         response;
+    const char  *title = _("Are you sure you want to delete the entries ?");
+
+    dialog = gtk_message_dialog_new (GTK_WINDOW (imap_dialog->dialog),
+                                     GTK_DIALOG_DESTROY_WITH_PARENT,
+                                     GTK_MESSAGE_QUESTION,
+                                     GTK_BUTTONS_CANCEL,
+                                     "%s", title);
+
+    gtk_dialog_add_button (GTK_DIALOG(dialog), _("_Delete"), GTK_RESPONSE_ACCEPT);
+
+    gtk_widget_grab_focus (gtk_dialog_get_widget_for_response (GTK_DIALOG(dialog), GTK_RESPONSE_ACCEPT));
+
+    response = gtk_dialog_run (GTK_DIALOG(dialog));
+
+    gtk_widget_destroy (dialog);
+
+    if (response == GTK_RESPONSE_ACCEPT)
+        return TRUE;
+    else
+        return FALSE;
+}
+
+static void
+gnc_imap_dialog_delete (ImapDialog *imap_dialog)
+{
+    GList            *list, *row;
+    GtkTreeModel     *model;
+    GtkTreeIter       iter;
+    GtkTreeSelection *selection;
+
+    model = gtk_tree_view_get_model (GTK_TREE_VIEW(imap_dialog->view));
+    selection = gtk_tree_view_get_selection (GTK_TREE_VIEW(imap_dialog->view));
+
+    list = gtk_tree_selection_get_selected_rows (selection, &model);
+
+    // Make sure we have some rows selected
+    if (g_list_length (list) == 0)
+        return;
+
+    // Are we sure we want to delete the entries
+    if (are_you_sure (imap_dialog) == FALSE)
+        return;
+
+    // reverse list
+    list = g_list_reverse (list);
+
+    for (row = g_list_first (list); row; row = g_list_next (row))
+    {
+	if (gtk_tree_model_get_iter (model, &iter, row->data))
+        {
```

Comment:
If you're writing a big nested scope with its own variables like this it means that it needs to be a separate function.